common/math/cubic_hermite_spline_test.cc

```diff
@@ -0,0 +1,76 @@
+
+#include "common/math/cubic_hermite_spline.hh"
+
+#include "gtest/gtest.h"
+
+namespace robot::math {
+
+TEST(CubicHermiteSplineTest, inputs_must_be_sorted) {
+    // Setup
+    const std::vector<double> ts{0.0, 1.0, 3.0, 2.0, 4.0};
+    const std::vector<double> xs{10.0, 20.0, 40.0, 30.0, 50.0};
+
+    // Action + Verification
+    EXPECT_THROW(CubicHermiteSpline(ts, xs), check_failure);
+}
+
+TEST(CubicHermiteSplineTest, query_after_bounds_throws) {
+    // Setup
+    const std::vector<double> ts{0.0, 1.0, 2.0, 3.0, 4.0};
+    const std::vector<double> xs{10.0, 20.0, 40.0, 30.0, 50.0};
+    const auto spline = CubicHermiteSpline(ts, xs);
+
+    // Action + Verification
+    EXPECT_THROW(spline(5.0), check_failure);
+}
+
+TEST(CubicHermiteSplineTest, query_before_bounds_throws) {
+    // Setup
+    const std::vector<double> ts{0.0, 1.0, 2.0, 3.0, 4.0};
+    const std::vector<double> xs{10.0, 20.0, 40.0, 30.0, 50.0};
+    const auto spline = CubicHermiteSpline(ts, xs);
+
+    // Action + Verification
```

Comment:
query_on_bounds doesn't throw test?

Reply:
There was a bug! thanks for suggesting this test.